src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -90,253 +139,3541 @@ static ColliderJntSphInit D_80A65F28 = {
+    func_800AA000(0.0f, 170, 10, 10);
+    this->postDrawFunc = NULL;
+}
+
+void EnHorse_StartBridgeJump(EnHorse* this, GlobalContext* globalCtx) {
+    this->postDrawFunc = EnHorse_BridgeJumpInit;
+    if (this->bridgeJumpIdx == 0) {
+        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(D_02002AC0);
+        gSaveContext.cutsceneTrigger = 1;
+    } else {
+        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(D_02000230);
+        gSaveContext.cutsceneTrigger = 1;
+    }
+}
+
+void EnHorse_BridgeJumpMove(EnHorse* this, GlobalContext* globalCtx) {
+    f32 interp;
+    f32 timeSq;
+
+    interp = this->bridgeJumpTimer / 30.0f;
+    timeSq = (this->bridgeJumpTimer * this->bridgeJumpTimer);
+
+    this->actor.world.pos.x = ((sBridgeJumps[this->bridgeJumpIdx].pos.x - this->bridgeJumpStart.x) * interp) + this->bridgeJumpStart.x;
+    this->actor.world.pos.z = ((sBridgeJumps[this->bridgeJumpIdx].pos.z - this->bridgeJumpStart.z) * interp) + this->bridgeJumpStart.z;
+
+    this->actor.world.pos.y = (this->bridgeJumpStart.y + (this->bridgeJumpYVel * this->bridgeJumpTimer) + (-0.4f * timeSq));
+
+    this->actor.world.rot.y = this->actor.shape.rot.y =
+        (sBridgeJumps[this->bridgeJumpIdx].angle + ((1.0f - interp) * this->bridgeJumpRelAngle));
+    this->skin.skelAnime.curFrame = 23.0f * interp;
+    SkelAnime_Update(&this->skin.skelAnime);
+    if (this->bridgeJumpTimer < 30) {
+        this->flags |= ENHORSE_FLAG_24;
+    }
+}
+
+void EnHorse_CheckBridgeJumpLanding(EnHorse* this, GlobalContext* globalCtx) {
+    this->actor.speedXZ = 8.0f;
+    this->skin.skelAnime.playSpeed = 1.5f;
+    if (SkelAnime_Update(&this->skin.skelAnime) != 0) {
+        this->flags &= ~ENHORSE_JUMPING;
+        this->actor.gravity = -3.5f;
+        this->actor.world.pos.y = sBridgeJumps[this->bridgeJumpIdx].pos.y;
+        func_80028A54(globalCtx, 25.0f, &this->actor.world.pos);
+        EnHorse_JumpLanding(this, globalCtx);
+        Audio_PlaySoundGeneral(NA_SE_EV_HORSE_LAND, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        func_800AA000(0.0f, 255, 10, 80);
+    }
+}
+
+void EnHorse_BridgeJump(EnHorse* this, GlobalContext* globalCtx) {
+    this->bridgeJumpTimer++;
+    if (this->bridgeJumpTimer < 30) {
+        EnHorse_BridgeJumpMove(this, globalCtx);
+        return;
+    }
+    EnHorse_CheckBridgeJumpLanding(this, globalCtx);
+}
+
+void EnHorse_Vec3fOffset(Vec3f* src, s16 yaw, f32 dist, f32 height, Vec3f* dst) {
+    dst->x = src->x + Math_SinS(yaw) * dist;
+    dst->y = src->y + height;
+    dst->z = src->z + Math_CosS(yaw) * dist;
+}
+
+s32 EnHorse_CalcFloorHeight(EnHorse* this, GlobalContext* globalCtx, Vec3f* pos, CollisionPoly** floorPoly, f32* floorHeight) {
+    s32 bgId;
+    f32 waterY;
+    WaterBox* waterBox;
+
+    *floorPoly = NULL;
+    *floorHeight = BgCheck_EntityRaycastFloor3(&globalCtx->colCtx, floorPoly, &bgId, pos);
+
+    if (*floorHeight == -32000.0f) {
```

Comment:
`BGCHECK_Y_MIN` is the define for this constant i believe, may be slightly wrong on the name